xtable-service/src/test/java/org/apache/xtable/service/TestConversionResource.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.apache.xtable.service.models.ConvertTableRequest;
+import org.apache.xtable.service.models.ConvertTableResponse;
+import org.apache.xtable.service.models.RestTargetTable;
+
+@ExtendWith(MockitoExtension.class)
+class TestConversionResource {
```

Comment:
Can we add an IT which writes hudi, iceberg and delta tables, bootstrap the server and run the conversion sync?  See IT tests in xtable-core.

Reply:
Yea i can try looking into this.

Reply:
@vinishjail97 I think this pr has gotten quite large, can we instead fast follow on another pr for just integration tests. My concern is that I will need to bring in some quarkus related integ tests deps which will potentially cause other issues. Currently haved added an intial set of untit tests and have manually ran the service.